tests/element_test_vectors.rs

```diff
@@ -717,40 +648,20 @@ mod non_blocking_native_element_tests {
         }
     }
 
-    #[test_resources("ion-tests/iontestdata/good/**/*.ion")]
-    #[test_resources("ion-tests/iontestdata/good/**/*.10n")]
-    fn native_good_roundtrip_text_binary(file_name: &str) {
-        good_roundtrip_text_binary(NonBlockingNativeElementApi, file_name)
-    }
-
-    #[test_resources("ion-tests/iontestdata/good/**/*.ion")]
-    #[test_resources("ion-tests/iontestdata/good/**/*.10n")]
-    fn native_good_roundtrip_binary_text(file_name: &str) {
-        good_roundtrip_binary_text(NonBlockingNativeElementApi, file_name)
-    }
-
-    #[test_resources("ion-tests/iontestdata/good/**/*.ion")]
-    #[test_resources("ion-tests/iontestdata/good/**/*.10n")]
-    fn native_good_roundtrip_text_pretty(file_name: &str) {
-        good_roundtrip_text_pretty(NonBlockingNativeElementApi, file_name)
-    }
-
-    #[test_resources("ion-tests/iontestdata/good/**/*.ion")]
-    #[test_resources("ion-tests/iontestdata/good/**/*.10n")]
-    fn native_good_roundtrip_pretty_text(file_name: &str) {
-        good_roundtrip_pretty_text(NonBlockingNativeElementApi, file_name)
-    }
-
-    #[test_resources("ion-tests/iontestdata/good/**/*.ion")]
-    #[test_resources("ion-tests/iontestdata/good/**/*.10n")]
-    fn native_good_roundtrip_pretty_binary(file_name: &str) {
-        good_roundtrip_pretty_binary(NonBlockingNativeElementApi, file_name)
-    }
-
-    #[test_resources("ion-tests/iontestdata/good/**/*.ion")]
-    #[test_resources("ion-tests/iontestdata/good/**/*.10n")]
-    fn native_good_roundtrip_binary_pretty(file_name: &str) {
-        good_roundtrip_binary_pretty(NonBlockingNativeElementApi, file_name)
+    good_round_trip! {
+        use NonBlockingNativeElementApi;
+        fn binary_compact(Format::Binary, Format::Text(TextKind::Compact));
+        fn binary_lines(Format::Binary, Format::Text(TextKind::Lines));
+        fn binary_pretty(Format::Binary, Format::Text(TextKind::Pretty));
+        fn compact_binary(Format::Text(TextKind::Compact), Format::Binary);
+        fn compact_lines(Format::Text(TextKind::Compact), Format::Text(TextKind::Lines));
+        fn compact_pretty(Format::Text(TextKind::Compact), Format::Text(TextKind::Pretty));
+        fn lines_binary(Format::Text(TextKind::Lines), Format::Binary);
+        fn lines_compact(Format::Text(TextKind::Lines), Format::Text(TextKind::Compact));
+        fn lines_pretty(Format::Text(TextKind::Lines), Format::Text(TextKind::Pretty));
+        fn pretty_binary(Format::Text(TextKind::Pretty), Format::Binary);
+        fn pretty_compact(Format::Text(TextKind::Pretty), Format::Text(TextKind::Compact));
+        fn pretty_lines(Format::Text(TextKind::Pretty), Format::Text(TextKind::Lines));
```

Comment:
It looks like this function list is copied verbatim in each `good_round_trip!` invocation; could it be moved to the macro body?

Reply:
It could, but I think I'd have to either write another macro (or pattern in the existing macro) or I'd have to write them in their expanded form into the macro.

I think this particular implementation hits a nice sweet spot by hiding the really repetitive stuff while still reading kind of like tests, and minimizing code without getting too complex with the macros.